### PR TITLE
add weighted quantiles

### PR DESCRIPTION
**Pull request by gidden**
# Description of PR

Adds a `df.quantile()` function which calculates (possibly weighted) quantiles for a variable across a population of models and scenarios.

If this feature is useful for others in the community, I'm happy to improve the PR, add a tutorial, etc.

cc @chrisroadmap

# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- ~[ ] Name of contributors Added to AUTHORS.rst~
- [x] Description in RELEASE_NOTES.md Added

# TODOS - issues

- [x] support custom names (currently put into `model`)
- [x] more flexible `weights` argument

## Adding to RELEASE_NOTES.md (remove section after adding to RELEASE_NOTES.md)

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

### CODE REVIEW

**Comment by danielhuppmann** [comment]
Thanks @gidden, this sounds really useful!

A few comments:
1. I developed a **Statistics** module for the SR15, see the [docs](https://pyam-iamc.readthedocs.io/en/stable/api/statistics.html) and a real-world use case [here](https://github.com/IAMconsortium/pyam/issues/581#issuecomment-921595487). The main advantage of the Statistics class is that you often want to do quantiles on groups of scenarios (e.g., the temperature categorization). If I understand the current implementation correctly, the method will always compute quantiles grouped by scenario names indexed by model, region and variable - we should probably anticipate a more flexible approach.
2. If you proceed with implementation in this way, I suggest to add that as method like `IamDataFrame.compute.quantiles()` to avoid too many methods directly on the **IamDataFrame** class. See the [docs](https://pyam-iamc.readthedocs.io/en/stable/api/compute.html).

**Comment by gidden** [comment]
@danielhuppmann - FYI - I am seeing test failures in `test_datareader.py` not related to this PR.

I'm also hoping to clean this up, time permitting, in the next days.

**Comment by danielhuppmann** [comment]
> @danielhuppmann - FYI - I am seeing test failures in test_datareader.py not related to this PR.

But all checks except one passed, so I guess that the Woldbank-server just had a hick-up in that second. Re-running the failed test might do the trick...

File: pyam/compute.py [hunk 1]
```diff
@@ -22,6 +22,71 @@ class IamComputeAccessor:
...
+        ----------
+        quantiles : collection
+            Group of quantile values to compute
+        weights : pd.Series, optional
+            Series indexed by `level`
+        level : collection, optional
+            The index columns to compute quantiles over
+        append : bool, optional
+            Whether to append computed timeseries data to this instance.
+        """
```
Comment (danielhuppmann) [inline pyam/compute.py:47]: Please add a **Returns** section

File: pyam/compute.py [hunk 2]
```diff
@@ -22,6 +22,71 @@ class IamComputeAccessor:
...
+        self_df = self._df
+        if len(self_df.variable) > 1:
+            raise ValueError(
+                "quantiles() currently supports only 1 variable, and this"
+                f"dataframe has {len(self_df.variable)}"
+            )
+        df = self_df.timeseries()
+        model = "unweighted" if weights is None else "weighted"  # can make this a kwarg
```
Comment (danielhuppmann) [inline pyam/compute.py:55]: I'm wondering whether "unweighted" is intuitive...
```suggestion
        model = "Quantiles" if weights is None else f"Quantiles (weighted by '{weights}'"
```

File: pyam/core.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 import logging
 import os
 import sys
+import wquantiles
```
Comment (danielhuppmann) [inline pyam/core.py:7]: This is probably a leftover?

File: pyam/compute.py [hunk 4]
```diff
@@ -22,6 +22,71 @@ class IamComputeAccessor:
...
+                wquantiles.quantile(df.loc[idx].values.T, w.values, q),
+                index=pd.Series(df.columns, name="year"),
+                name="value",
+            )
+            kwargs = {idxs.names[i]: idx[i] for i in range(len(idx))}
+            dfs.append(
+                IamDataFrame(
+                    data,
+                    model=model,
+                    scenario=f"quantile_{q}",  # can make this a kwarg
```
Comment (danielhuppmann) [inline pyam/compute.py:82]: If you want to plot directly, using the quantile values directly would look nicer...
```suggestion
                    scenario=q,
```

**Review by danielhuppmann — COMMENTED** [review]
Looks good, a few minor suggestions inline!

File: tests/test_feature_quantiles.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+from pyam import IamDataFrame
+import pytest
+from pyam.testing import assert_iamframe_equal
+import pandas as pd
+
+
+def test_qunatile_one_variable(test_pd_df):
```
Comment (danielhuppmann) [inline tests/test_feature_quantiles.py:7]:
```suggestion
def test_quantile_one_variable(test_pd_df):
```

File: tests/test_feature_quantiles.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        ),
+        model="Quantiles",
+        region="World",
+        variable="Primary Energy",
+        unit="EJ/yr",
+    )
+    assert_iamframe_equal(exp, obs)
+
+
+def test_quantile_multiple_variables(test_pd_df):
```
Comment (danielhuppmann) [inline tests/test_feature_quantiles.py:32]:
```suggestion
def test_quantile_missing_variable(test_pd_df):
```

**Review by danielhuppmann — APPROVED** [review]
Three more minor fixes, then good to be merged. Thanks for adding a tutorial!

**Comment by gidden** [comment]
all committed, should be g2g @danielhuppmann ! Thanks for the review!

**Comment by danielhuppmann** [comment]
Release notes are still missing - pretty please!

**Comment by gidden** [comment]
Release notes updated and issues made @danielhuppmann

**Comment by danielhuppmann** [comment]
Thanks @gidden!

**Pull request merged**